bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/SurePetcareAPIHelper.java

```diff
@@ -0,0 +1,481 @@
+        control.lockingModeId = newLockingModeId;
+        String ctrlurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/control";
+        setDataThroughApi(ctrlurl, HttpMethod.PUT, control);
+
+        // now we're fetching the new state back for the cache
+        String devurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/status";
+        SurePetcareDeviceStatus newStatus = SurePetcareConstants.GSON.fromJson(getDataFromApi(devurl),
+                SurePetcareDeviceStatus.class);
+        device.status.assign(newStatus);
+    }
+
+    /**
+     * Updates the device led mode through an API call to the Sure Petcare API.
+     *
+     * @param device the device
+     * @param newLedModeId the id of the new led mode
+     * @throws SurePetcareApiException
+     */
+    public synchronized void setDeviceLedMode(SurePetcareDevice device, Integer newLedModeId)
+            throws SurePetcareApiException {
+        // post new JSON control structure to API
+        SurePetcareDeviceControl control = new SurePetcareDeviceControl();
+        control.ledModeId = newLedModeId;
+        String ctrlurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/control";
+        setDataThroughApi(ctrlurl, HttpMethod.PUT, control);
+
+        // now we're fetching the new state back for the cache
+        String devurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/status";
+        SurePetcareDeviceStatus newStatus = SurePetcareConstants.GSON.fromJson(getDataFromApi(devurl),
+                SurePetcareDeviceStatus.class);
+        device.status.assign(newStatus);
+    }
+
+    /**
+     * Updates all curfews through an API call to the Sure Petcare API.
+     *
+     * @param device the device
+     * @param curfewList the list of curfews
+     * @throws SurePetcareApiException
+     */
+    public synchronized void setCurfews(SurePetcareDevice device, SurePetcareDeviceCurfewList curfewList)
+            throws SurePetcareApiException {
+        // post new JSON control structure to API
+        SurePetcareDeviceControl control = new SurePetcareDeviceControl();
+        control.curfewList = curfewList.compact();
+        String ctrlurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/control";
+        setDataThroughApi(ctrlurl, HttpMethod.PUT, control);
+
+        // now we're fetching the new state back for the cache
+        String devurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/control";
+        SurePetcareDeviceControl newControl = SurePetcareConstants.GSON.fromJson(getDataFromApi(devurl),
+                SurePetcareDeviceControl.class);
+        if (newControl != null) {
+            newControl.curfewList = newControl.curfewList.order();
+        }
+        device.control = newControl;
+    }
+
+    /**
+     * @return true, if the API is connected and successfully authenticated.
+     */
+    // public final boolean isOnline() {
+    // return online;
+    // }
```

Comment:
Please remove if you aren't going to use.

Reply:
done